api/envoy/extensions/filters/http/waf/v3/waf.proto

```diff
@@ -0,0 +1,223 @@
+
+message WAF {
+
+// WAF filter provides ability to allow, block, redirect and modify HTTP
+// requests based on rules. The rules are evaluated in order in which they
+// are present.
+//
+// Here is an example of WAF filter configuration:
+//
+//  .. code-block:: yaml
+//
+//  executor:
+//	name: envoy.priority
+//
+//  policy_groups:
+//  - name: grp_1
+//    policies:
+//    - name: block-admin-pages
+//      match:
+//        header:
+//          name: ":path"
+//          prefix: "/admin"
+//      watcher:
+//        - name: envoy.watcher
+//          action:
+//            reset: truek
+//    - name: ofac-white-list
+//      match:
+//        sourceIp:
+//          cidrs:
+//          - addressPrefix: 2.144.136.0
+//            prefixLength: 21
+//          - addressPrefix: 2.144.144.0
+//            prefixLength: 20
+//          - addressPrefix: 2.144.160.0
+//            prefixLength: 19
+//      watcher:
+//        - name: envoy.watcher
+//          action:
+//            nop:true
+//    - name: ofac-black-list
+//      match:
+//        andRules:
+//          rules:
+//          - header:
+//              name: ":path"
+//              prefix: "/ofac-page"
+//              invertMatch: true
+//          - sourceIp:
+//              cidrs:
+//              - addressPrefix: 2.144.136.0
+//                prefixLength: 21
+//              - addressPrefix: 2.144.144.0
+//                prefixLength: 20
+//              - addressPrefix: 2.144.160.0
+//                prefixLength: 19
+//      watcher:
+//        - name: envoy.watcher
+//          action:
+//            http_redirect:
+//              query_params: >
+//                "name1=value1&ref=%REQ(X-REQUEST-ID)%&name3=val3"
+//              response_code: "302"'
+//              host: "www.acme.com"'
+//      rule_match_cb: envoy.until_first_match
+//  - name: grp_2
+//    policies:
+//	  - name: ciruit-breaker
+//    	match:
+//      	header: {name: "path", prefix_match: "/signup"}
+//    	watcher:
+//      	name: custom.circuit_breaker
+//      	config:
+//        	type : gradient2
+//        	action: drop
+//	 - name: bot-analyzer
+//     match:
+//       header: {name: "path", prefix_match: "/"}
+//     watcher:
+//       name: custom.bot_analyzer
+//   rule_match_cb: envoy.all
+
+message WAF {
+  message PolicySet {
+    repeated Policy policies = 1[(validate.rules).repeated .min_items = 1];
+  }
+
+  message PolicyGroup {
+    // Name of the Policy Group.
+    string name = 1 [(validate.rules).string.pattern = "^[a-zA-Z0-9]([-a-zA-Z0-9]*[a-zA-Z0-9])?$"];
+    PolicySet policies = 2;
+    RuleMatchCB callback = 3;
+  }
+
+  repeated PolicyGroup policy_groups = 1[(validate.rules).repeated .min_items = 1];
+  Executor executor = 2;
+ }
+
+ // Rule defines a match condition and an associated action.
+ message Policy {
+  string name = 1 (validate.rules).string.pattern = "^[a-zA-Z0-9]([-a-zA-Z0-9]*[a-zA-Z0-9])?$"];
+  Match match = 2;
+  Watcher watcher = 3;
```

Comment:
I'm having a hard time trying to distinguish match and watcher in my head. Can you write a sections in `docs/root/configuration` that explains in detail the model for WAF and how these differ?